### ISSUE TITLE
to_dataframe() on allow_none columns 

### ISSUE THREAD

**Issue by bsnacks000** [bsnacks000/binx#17]
* binx version:
* Python version:
* Operating System:

### Description

When allow_none is set to true, the to_dataframe() method must be able to handle None types 

### What I Did

```
File "/home/bsnacks000/py_projects/bema/tests/test_collections.py", line 206, in test_bestrejectedmodel_collection_loads_retrofit_fixture
    df = b.to_dataframe()
  File "/home/bsnacks000/py_projects/bema/venv/lib/python3.5/site-packages/binx/collection.py", line 379, in to_dataframe
    return self._dataframe_with_dtypes(self.data)
  File "/home/bsnacks000/py_projects/bema/venv/lib/python3.5/site-packages/binx/collection.py", line 286, in _dataframe_with_dtypes
    df_data[col] = pd.Series(col_data[col], dtype=dtype)
  File "/home/bsnacks000/py_projects/bema/venv/lib/python3.5/site-packages/pandas/core/series.py", line 275, in __init__
    raise_cast_failure=True)
  File "/home/bsnacks000/py_projects/bema/venv/lib/python3.5/site-packages/pandas/core/series.py", line 4114, in _sanitize_array
    subarr = _try_cast(data, False)
  File "/home/bsnacks000/py_projects/bema/venv/lib/python3.5/site-packages/pandas/core/series.py", line 4053, in _try_cast
    copy=copy)
  File "/home/bsnacks000/py_projects/bema/venv/lib/python3.5/site-packages/pandas/core/dtypes/cast.py", line 1257, in construct_1d_ndarray_preserving_na
    subarr = np.array(values, dtype=dtype, copy=copy)
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'

```

**Comment by bsnacks000** [comment]
merged into development branch for 0.3

**Comment by bsnacks000** [comment]
merged into development for 0.3
